plugins/analytics/public/packettotal.py

```diff
@@ -0,0 +1,147 @@
+from core.observables import Hostname, Ip, Hash, Url, Text
+from core.config.config import yeti_config
+
+class PacketTotalAPI(object):
+    """Base class for querying the PacketTotal API."""
+    settings = {
+        'packettotal_api_key': {
+            'name': 'PacketTotal API Key',
+            'description': 'API Key provided by packettotal.com'
+        }
+    }
+
+    API = 'https://api.packettotal.com/v1/'
+
+    @staticmethod
+    def search(observable, api_key):
+        #"""
+        #:param observable: The extended observable class
+        #:param api_key: The api key obtained from PacketTotal
+        #:return:  packettotal json response or None if error
+        #"""
+
+        params = {'query': observable.value}
+
+        try:
+            res = requests.get(
+                '{}search'.format(PacketTotalAPI.API),
+                headers={'x-api-key': api_key},
+                params=params,
+                verify=False,
+                proxies=yeti_config.proxy
+            )
+
+            if res.ok:
+                return res.json()
+            else:
+                return None
+
+        except Exception as e:
+            logging.error('Exception while getting packettotal report {}'.format(e.message))
+            return None
+
+
+    @staticmethod
+    def fetch_analysis(pcap_id, api_key):
+        #"""
+        #:param observable: The extended observable class
+        #:param api_key: The api key obtained from PacketTotal
+        #:return:  packettotal json response or None if error
+        #"""
+
+        try:
+            res = requests.get(
+                '{}pcaps/{}/analysis'.format(PacketTotalAPI.API, pcap_id),
+                headers={'x-api-key': api_key},
+                verify=False,
+                proxies=yeti_config.proxy
+            )
+
+            if res.ok:
+                return res.json()
+            else:
+                return None
+
+        except Exception as e:
+            logging.error('Exception while getting packettotal report {}'.format(e.message))
+            return None
+
+
+class PacketTotalQuery(PacketTotalAPI, OneShotAnalytics):
+    default_values = {
+        'name': 'PacketTotal',
+        'description': 'Perform a PacketTotal query.',
+    }
+
+    ACTS_ON = ['Ip', 'Hostname', 'Hash', 'Url']
+
+    @staticmethod
+    def analyze(observable, results):
+        links = set()
+        pcap_hits = PacketTotalAPI.search(observable, results.settings['packettotal_api_key'])
+        result = {}
+
+        if pcap_hits:
+            for result in pcap_hits.get('results', []):
+                pcap_id = result.get('id','')
+                pcap_analysis = PacketTotalAPI.fetch_analysis(pcap_id,  results.settings['packettotal_api_key'])
+                if pcap_analysis:
+                    analysis_url = Text.get_or_create(
+                        value = 'https://packettotal.com/app/analysis?id={pcap_id}'.format(pcap_id=pcap_id)
```

Comment:
```suggestion
                        value='https://packettotal.com/app/analysis?id={pcap_id}'.format(pcap_id=pcap_id)
```

This is a keyword arg, so no space after / before the `=` sign :)